client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDisplay/AllAttemptsDisplay.tsx

```diff
@@ -0,0 +1,155 @@
+import { FC, useState } from 'react';
+import { defineMessages } from 'react-intl';
+import { Slider, Typography } from '@mui/material';
+import { QuestionType } from 'types/course/assessment/question';
+import {
+  AllAnswerDetails,
+  QuestionDetails,
+} from 'types/course/statistics/assessmentStatistics';
+
+import { workflowStates } from 'course/assessment/submission/constants';
+import Accordion from 'lib/components/core/layouts/Accordion';
+import Link from 'lib/components/core/Link';
+import useTranslation from 'lib/hooks/useTranslation';
+import { formatLongDateTime } from 'lib/moment';
+
+import AnswerDetails from '../AnswerDetails/AnswerDetails';
+
+interface Props {
+  allAnswers: AllAnswerDetails<keyof typeof QuestionType>[];
+  question: QuestionDetails<keyof typeof QuestionType>;
+  questionNumber: number;
+  submissionEditUrl: string;
+}
+
+const translations = defineMessages({
+  questionTitle: {
+    id: 'course.assessment.statistics.questionTitle',
+    defaultMessage: 'Question {index}',
+  },
+  pastAnswerTitle: {
+    id: 'course.assessment.statistics.pastAnswerTitle',
+    defaultMessage: 'Submitted At: {submittedAt}',
+  },
+  mostRecentAnswer: {
+    id: 'course.assessment.statistics.mostRecentAnswer',
+    defaultMessage: 'Current Answer (Attempting)',
+  },
+  submissionPage: {
+    id: 'course.assessment.statistics.submissionPage',
+    defaultMessage: 'Go to Answer Page',
+  },
+});
+
+// only used as a dummy buffer time if the submission's state is attempting
+// this is due to the time record for attempting answer being lower than any other
+// states' answer, while it's actually the latest version.
+const BUFFER_TIME = 100;
```

Comment:
This is a monkeypatch, don't modify the createdAt like that.  
Wouldn't pushing the currentAnswer to the back of the sortedAnswers (as you've done in line 70) should be sufficient to "order" the attempts?

Either that, or simply set the "currentAnswer" in the attempting workflowState "submitted timestamp" to the current datetimestamp now.

Reply:
this is not for ordering actually (if you noticed inside this file, I intentionally took out the currentAnswer from the array then append it to the back of the array, that has been done)

this is done because in the slider, since we need to show the leftmost and rightmost timestamp (which is based on when the said answer is submitted), and that the rightmost one might not have its submitted_at filled in, then I did this patching in order to have the rightmost timestamp in slider not blank

Note: I have recently decided to only display the contents of the page if the submission is either graded or published (so actually this BUFFER_TIME is not needed anymore), though I also realise this might be kept in case in the future, we want to also display the past answers in statistics also for other workflow states (attempting)

Another way to mitigate this issue is not to use SubmittedAt for slider display, but instead just use "Most Recent Answer" and "Initial Answer". Please let me know what you prefer then I'll do the implementation accordingly

Reply:
btw, after deciding on only showing for Graded and Published state, I decided to remove this line (and also all the consequent logics that needs to check whether the state is attempting)

Reply:
Preferably squash/fixup.